ateam_kenobi/src/kenobi_node.cpp

```diff
@@ -0,0 +1,266 @@
+      blue_robots_subscriptions_,
+      Topics::kBlueTeamRobotPrefix,
+      10,
+      &KenobiNode::blue_robot_state_callback,
+      this);
+
+    create_indexed_subscribers<ateam_msgs::msg::RobotState>(
+      yellow_robots_subscriptions_,
+      Topics::kYellowTeamRobotPrefix,
+      10,
+      &KenobiNode::yellow_robot_state_callback,
+      this);
+
+    create_indexed_publishers<ateam_msgs::msg::RobotMotionCommand>(
+      robot_commands_publishers_, Topics::kRobotMotionCommandPrefix,
+      rclcpp::SystemDefaultsQoS(), this);
+
+    ball_subscription_ = create_subscription<ateam_msgs::msg::BallState>(
+      std::string(Topics::kBall),
+      10,
+      std::bind(&KenobiNode::ball_state_callback, this, std::placeholders::_1));
+
+    world_publisher_ = create_publisher<ateam_msgs::msg::World>(
+      "~/world",
+      rclcpp::SystemDefaultsQoS());
+
+    field_subscription_ = create_subscription<ateam_msgs::msg::FieldInfo>(
+      std::string(Topics::kField),
+      10,
+      std::bind(&KenobiNode::field_callback, this, std::placeholders::_1));
+
+    timer_ = create_wall_timer(10ms, std::bind(&KenobiNode::timer_callback, this));
+
+    RCLCPP_INFO(get_logger(), "Kenobi node ready.");
+  }
+
+private:
+  World world_;
+  visualization::OverlayPublisher overlay_publisher_;
+  visualization::PlayInfoPublisher play_info_publisher_;
+  PlaySelector play_selector_;
+  InPlayEval in_play_eval_;
+  rclcpp::Subscription<ateam_msgs::msg::BallState>::SharedPtr ball_subscription_;
+  std::array<rclcpp::Subscription<ateam_msgs::msg::RobotState>::SharedPtr,
+    16> blue_robots_subscriptions_;
+  std::array<rclcpp::Subscription<ateam_msgs::msg::RobotState>::SharedPtr,
+    16> yellow_robots_subscriptions_;
+  rclcpp::Subscription<ateam_msgs::msg::FieldInfo>::SharedPtr
+    field_subscription_;
+  std::array<rclcpp::Publisher<ateam_msgs::msg::RobotMotionCommand>::SharedPtr,
+    16> robot_commands_publishers_;
+
+  ateam_common::GameControllerListener game_controller_listener_;
+
+  rclcpp::Publisher<ateam_msgs::msg::World>::SharedPtr world_publisher_;
+
+  rclcpp::TimerBase::SharedPtr timer_;
+
+  void blue_robot_state_callback(
+    const ateam_msgs::msg::RobotState::SharedPtr robot_state_msg,
+    int id)
+  {
+    const auto are_we_blue = game_controller_listener_.GetTeamColor() ==
+      ateam_common::TeamColor::Blue;
+    auto & robot_state_array = are_we_blue ? world_.our_robots : world_.their_robots;
+    robot_state_callback(robot_state_array, id, robot_state_msg);
+  }
+
+  // TODO(CAVIDANO): REMOVE THE THEIR ROBOTS HERE THIS SHOULD NOT ASSIGN ANYTHING IN THE NOT CASE
+  // AS IT ALSO CATCHES UNKOWN TEAM
+  void yellow_robot_state_callback(
+    const ateam_msgs::msg::RobotState::SharedPtr robot_state_msg,
+    int id)
+  {
+    const auto are_we_yellow = game_controller_listener_.GetTeamColor() ==
+      ateam_common::TeamColor::Yellow;
+    auto & robot_state_array = are_we_yellow ? world_.our_robots : world_.their_robots;
```

Comment:
Change at some point for lines 130 and 142 to address case of Unknown team color (following Collin's TODO)